lib/index.js

```diff
@@ -252,10 +276,25 @@ Validator.prototype._getMessage = function (ruleName) {
   return this.options.messages[ruleName] || ((defaultValidations.hasOwnProperty(ruleName)) ? defaultValidations[ruleName].message : defaultValidations['default'].message);
 };
 
-Validator.prototype._check = function (field, rule) {
-  if(_.isRegExp(rule)) return rule.test(field.value);
+Validator.prototype._check = function (fieldName, value, validation, serial, context) {
+  return function (callback) {
+    var setValidity = function (valid) {
+      var current = (serial === context.model.get(fieldName + '.serial'));
+      if (current) {
+        if (!valid) context.model.push(fieldName + '.messages', validation.message);
+        context._setValidity(fieldName, valid);
+      }
+      
+      // If its not the most recent check OR validation is invalid we pass error true
+      // to fire async callback. Only if it's actually the validation we care about AND it's valid
+      // we pass null (meaning no error).
+      callback(current ? (valid ? null : true) : true, valid);
```

Comment:
Nested conditional statements like this are quite hard to read. In this specific scenario, I actually think you can make it much simpler too:
`callback(current && valid ? null : true, valid);`